packages/vscode-extension/src/treeview/environmentTreeItem.ts

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as util from "util";
+import * as vscode from "vscode";
+
+import { LocalEnvironmentName, SubscriptionInfo } from "@microsoft/teamsfx-api";
+
+import { AppStudioLogin } from "../commonlib/appStudioLogin";
+import AzureAccountManager from "../commonlib/azureLogin";
+import { signedIn } from "../commonlib/common/constant";
+import { ext } from "../extensionVariables";
+import {
+  getM365TenantFromEnv,
+  getProvisionSucceedFromEnv,
+  getResourceGroupNameFromEnv,
+  getSubscriptionInfoFromEnv,
+  isExistingTabApp,
+  isSPFxProject,
+} from "../utils/commonUtils";
+import { localize } from "../utils/localizeUtils";
+import { DynamicNode } from "./dynamicNode";
+
+enum EnvInfo {
+  Local = "local",
+  LocalForExistingApp = "local-existing-app",
+  RemoteEnv = "environment",
+  ProvisionedRemoteEnv = "environment-provisioned",
+}
+
+interface accountStatus {
+  isOk: boolean;
+  warnings: string[];
+}
+
+const provisionedIcon = new vscode.ThemeIcon("folder-active");
+const nonProvisionedIcon = new vscode.ThemeIcon("symbol-folder");
+const warningIcon = new vscode.ThemeIcon("warning");
+const subscriptionIcon = new vscode.ThemeIcon("key");
+const resourceGroupIcon = new vscode.ThemeIcon("symbol-method");
+
+export class EnvironmentNode extends DynamicNode {
+  constructor(public identifier: string) {
+    super(identifier, vscode.TreeItemCollapsibleState.None);
+  }
+
+  public async getChildren(): Promise<DynamicNode[] | undefined | null> {
+    const children: DynamicNode[] = [];
+    if (this.identifier !== LocalEnvironmentName) {
+      // check account status
+      const accountStatus = await this.checkAccountForEnvironment(this.identifier);
+      if (!accountStatus.isOk) {
+        const warningNode = new WarningNode(this.identifier, accountStatus.warnings);
+        children.push(warningNode);
+      }
+      // show subscription
+      const subscriptionInfo = await getSubscriptionInfoFromEnv(this.identifier);
+      if (subscriptionInfo) {
+        const subscriptionNode = new SubscriptionNode(this.identifier, subscriptionInfo);
+        children.push(subscriptionNode);
+      }
+    }
+    return children;
+  }
+
+  public async getTreeItem(): Promise<vscode.TreeItem> {
+    const envInfo = await this.getCurrentEnvInfo(this.identifier);
+    const isSpfxProject = await isSPFxProject(ext.workspaceUri.fsPath);
```

Comment:
seems the `await` is unnecessary.

Reply:
do we need to check if `ext.workspaceUri` is valid here?

Reply:
Thanks, fixed this `await` code.
The `ext.workspaceUri` is guaranteed in `EnvironmentTreeViewProvider:reloadEnvironments()` so it should always be valid here.

Reply:
This makes sense and it's fixed.